Dfe.Data.SearchPrototype.Data/PostcodeLookupService.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Net.Http.Json;
+using System.Text;
+using System.Text.Json;
+using Dfe.Data.SearchPrototype.Data.Models;
+
+namespace Dfe.Data.SearchPrototype.Data;
+
+public class PostcodeLookupService
+{
+    private IHttpClientFactory _httpClientFactory;
+    public PostcodeLookupService(IHttpClientFactory httpClientFactory)
+    {
+        _httpClientFactory = httpClientFactory;
+    }
+
+    public async Task<IEnumerable<GeoLocation?>?> LookupPostcodes(IEnumerable<string> postcodes)
```

Comment:
`minor` could this return an empty list than `IEnumerable<T>?` to avoid propagating null checks where it's called.

`result?.Result..... ?? []`

Reply:
yes, modified to return an empty collection.
the null checks on the result are allowing for the possible null values returned by the API